packages/block-library/src/post-comment-avatar/edit.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	InspectorControls,
+	__experimentalUseBorderProps as useBorderProps,
+	__experimentalImageSizeControl as ImageSizeControl,
+	useBlockProps,
+} from '@wordpress/block-editor';
+import { PanelBody, ResizableBox } from '@wordpress/components';
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+import { __, isRTL } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+
+export default ( { attributes, context: { commentId }, setAttributes } ) => {
+	const { className, style, height, width, alt, url } = attributes;
+
+	const { comment } = useSelect(
+		( select ) => {
+			const { getEntityRecord } = select( coreStore );
+			return {
+				comment: getEntityRecord( 'root', 'comment', commentId ),
+			};
+		},
+		[ commentId ]
+	);
+	const borderProps = useBorderProps( attributes );
+	const authorAvatarUrls = comment?.author_avatar_urls; // eslint-disable-line camelcase
+	const avatarUrls = authorAvatarUrls
+		? Object.values( authorAvatarUrls )
+		: null;
+	const [ { defaultWidth, defaultHeight }, setDefaultSize ] = useState( {} );
+
+	// Set default widht, height and url on first render.
+	useEffect( () => {
+		setDefaultSize( { defaultWidth: width, defaultHeight: height } );
+	}, [] );
+
+	useEffect( () => {
+		setAttributes( {
```

Comment:
When we `setAttributes` we also add a step in  `history` (undo, redo). Since `avatarUrls` is an array will always be a new reference, so it causes a history trap. You can check this by inserting the block and try to `undo`. If you needed only this `avatarUrls[ avatarUrls.length - 1 ]` which is a `string`, this should be the dependency.